test/check-machines-lifecycle

```diff
@@ -272,6 +272,64 @@ class TestMachinesLifecycle(VirtualMachinesCase):
         b.wait_in_text(".pf-c-popover", "VM subVmTest2 failed to start")
         b.click('#vm-subVmTest2-system-state button[aria-label=Close]')
 
+    def testVmFilterSettingsOnNavigate(self):
```

Comment:
This test does not really test what this commit introduces. Instead you want to check that the URL is affected when you apply some status filter, and vice versa, when you apply a status filter, and refresh and filter is persisting.

Use something like:   b.wait(lambda: "vm?name=test%25&connection=syste,&status=shut-off" in b.eval_js("window.location.href"))
for testing the URL.

Reply:
Understood. Thanks! Last time my thought was the URL determines the state and the state in turn determines which vms are listed. So checking the vms listed also checks whether the URL works in an end to end way.

But I think I understand the subtle difference now.

URL -> State (The commit introduces this. I should check this)
State -> VMs listed (It was already implemented. No need to check this)

Fixed it!

I'm also checking for only one filter option to avoid repetition this time. Assuming that if it works for one (e.g. "running") it'd also work for others too (e.g. "shut off"). Happy to get any feedback on that :)